rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,611 @@
+          ),
+        },
+      ],
+    };
+  });
+
+  return (
+    <View>
+      <Animated.View style={[styles.block, shakeAnimationStyle]} />
+      <Button text="Shake" onPress={() => {
+        shakeAnimation.value = withTiming(100, { // Change the value of shakeAnimation to 100
+          duration: motionToken.duration.duration3, // Define the duration it takes to change the value of shakeAnimation to 100
+          easing: motionToken.easing.standard.effective, // Define the easing of the animation
+        });
+      }}
+      />
+    </View>
+  );
+}
+```
+
+- React Native's **Animated API** works on the **same principle as React Native**; you define your **animations** **in JS** and they **execute** on the **Native** **platforms**.
+- The **downside** of this approach is that it still needs to **communicate with** the Native realm using **the bridge** which can lead to **dropped frames** in complex animations in cases where the **bridge is already choked up**.
+- **React native reanimated** (v2) aims to provide ways of **offloading animation** and event handling logic off of the JavaScript thread and onto the **UI thread**.
+- This is achieved by defining **Reanimated worklets** – chunks of JavaScript code that can be moved to a **separate JavaScript VM** and executed synchronously on the UI thread. This makes it possible to respond to touch events immediately and update the UI within the same frame when the event happens **without worrying about** the load that is put on the **main JavaScript thread**. 
+- In the above example, we are using `useAnimatedStyle` to create a worklet that will be executed on the UI thread and has a **shared value** `translateXOffset` that is shared by JS as well as native realms.
+- React Native reanimated is able to achieve this using React Native's `TurboModules` feature.
+- Since **we want to leverage these benefits** of **React Native Reanimated**, **we will be using it** for our native **animations on Blade**.
+
+
+# Drawbacks/Constraints
+### Library specific tokens
+Consumers would need to be aware that certain tokens are library specific and would not work with other libraries
+- For web, we expose `duration` as a `string` of *milliseconds* which works well with CSS's `transition-duration` property but will not work with Framer motion since it requires a `number` in *seconds*
+- For web, we expose `easing` as a `string` of `cubic-bezier(..)` which works well with CSS's `transition-timing-function` property but will not work with Framer motion since it requires either,
+  - The name of an existing easing function.
+  - An array of four numbers to define a cubic bezier curve.
+  - An easing function, that accepts and returns a value 0-1.
+- For native, we expose `easing` as a function of `Easing.bezier(..)` imported from `react-native-reanimated` which won't work when used with React Native's `Animated` API.
+
+### React Native `v0.62`
```

Comment:
wouldn't this be a biggest problem? since we now have React 18 out which also exposes certain things we might want to use and if we are limited to this version then that might also cause issue?

Reply:
Not limited to this version. Any version greater than or equal to 0.62. I'll rephrase it to make it clear